lib/sitemap.js

```diff
@@ -30,338 +32,10 @@ exports.buildSitemapIndex = buildSitemapIndex;
  * @param   {String}        conf.xmlNs
  * @return  {Sitemap}
  */
-function createSitemap(conf) {
-  return new Sitemap(conf.urls, conf.hostname, conf.cacheTime, conf.xslUrl, conf.xmlNs);
+function createSitemap (conf) {
+  return new Sitemap(conf.urls, conf.hostname, conf.cacheTime, conf.xslUrl, conf.xmlNs)
 }
 
-function safeUrl(conf) {
-  var loc = conf['url'];
-  if (!conf['safe']) {
-    var url_parts = urlparser.parse(conf['url']);
-    if (!url_parts['protocol']) {
-      throw new err.NoURLProtocolError();
-    }
-
-    loc = htmlEscape(conf['url']);
-  }
-  return loc;
-}
-
-function safeDuration(duration) {
-  if (duration < 0 || duration > 28800) {
-    throw new err.InvalidVideoDuration();
-  }
-
-  return duration
-}
-
-var allowDeny = /^allow|deny$/
-var validators = {
-  'price:currency': /^[A-Z]{3}$/,
-  'price:type': /^rent|purchase|RENT|PURCHASE$/,
-  'price:resolution': /^HD|hd|sd|SD$/,
-  'platform:relationship': allowDeny,
-  'restriction:relationship': allowDeny
-}
-
-function attrBuilder(conf, keys) {
-  if (typeof keys === 'string') {
-    keys = [keys]
-  }
-
-  var attrs = keys.reduce((attrString, key) => {
-    if (conf[key] !== undefined) {
-      var keyAr = key.split(':')
-      if (keyAr.length !== 2) {
-        throw new err.InvalidAttr(key)
-      }
-
-      if (validators[key] && !validators[key].test(conf[key])) {
-        throw new err.InvalidAttrValue(key, conf[key], validators[key])
-      }
-      attrString += ' ' + keyAr[1] + '="' + conf[key] + '"'
-    }
-
-    return attrString
-  }, '')
-
-  return attrs
-}
-
-/**
- * Item in sitemap
- */
-function SitemapItem(conf) {
```

Comment:
I have split Sitemap Item out into a separate file as the single file was getting fairly large and unwieldy.